common/flatpak-instance.c

```diff
@@ -482,32 +482,8 @@ flatpak_instance_is_running (FlatpakInstance *self)
 {
   FlatpakInstancePrivate *priv = flatpak_instance_get_instance_private (self);
 
-  if (priv->dir)
-    {
-      g_autofree char *ref_file = g_strconcat (priv->dir, "/.ref", NULL);
-      struct stat statbuf;
-      struct flock l = {
-        .l_type = F_WRLCK,
-        .l_whence = SEEK_SET,
-        .l_start = 0,
-        .l_len = 0
-      };
-      glnx_autofd int lock_fd = open (ref_file, O_RDWR | O_CLOEXEC);
-      if (lock_fd != -1 &&
-          fstat (lock_fd, &statbuf) == 0 &&
-          /* Only gc if created at least 3 secs ago, to work around race mentioned in flatpak_run_allocate_id() */
-          statbuf.st_mtime + 3 < time (NULL) &&
-          fcntl (lock_fd, F_GETLK, &l) == 0 &&
-          l.l_type == F_UNLCK)
-        {
-          /* The instance is not used, remove it */
-          g_debug ("Cleaning up unused container id %s", priv->id);
-          glnx_shutil_rm_rf_at (-1, priv->dir, NULL, NULL);
-
-          g_clear_pointer (&priv->dir, g_free);
-        }
-    }
+  g_autofree char *proc_file = g_strdup_printf ("/proc/%d/cmdline", priv->pid);
 
-  return priv->dir != NULL;
+  return g_file_test (proc_file, G_FILE_TEST_EXISTS);
```

Comment:
I think kill(pid, 0) is a better check for this:
```     
  If sig is 0, then no signal is sent, but existence and permission checks are still performed; this can be used to check for the existence of a process ID or process group ID that the caller is permitted to signal.
```